tools/hidl/hidlcomp.cpp

```diff
@@ -395,12 +395,42 @@ void indentOutf1(int inc, const char* fmt, ...)
 
 void validateProfileExecutionOptions(std::string &options)
 {
-    size_t pos = options.find_first_of(',');
-    if (pos == std::string::npos)
+    //
+    // Expected format:
+    //   options     ::= ("s" | "ms" | "us" | "ns") "," <bucketLimit> ["," <bucketLimit>]*
+    //   bucketLimit ::= digit+
+    // Note, caller must remove any spaces
+
+    //
+    // Split the option string up
+    std::vector<std::string> optionValues;
+    size_t start;
+    size_t end = 0;
+    while ((start = options.find_first_not_of(',', end)) != std::string::npos) {
+        end = options.find(',', start);
+        optionValues.push_back(options.substr(start, end - start));
+    }
+
+    //
+    // Must be at least 2, units and a bucket
+    if (optionValues.size() < 2)
         throw "Execution profiling option must define at least one bucket";
-    std::string units = options.substr(0, pos);
-    if (units != "s" && units != "ms" && units  != "us" && units != "ns")
+
+    //
+    // First entry is units
+    if (optionValues[0] != "s" && optionValues[0] != "ms" && optionValues[0]  != "us" && optionValues[0] != "ns")
         throw "Execution profiling units must be s, ms, us, or ns";
+
+    //
+    // Remaining entries must be non-zero numbers increasing in value
+    int value, maxValue = 0;
+    for (unsigned i=1; i<optionValues.size(); ++i)
+    {
+        value = std::stoi(optionValues[i]);
```

Comment:
picky: easier to read if value is also declared here, rather than wondering why value might be used after the loop, and is it a problem it isn't initialised.